### ISSUE TITLE
TA build linker settings to not link in unused functions

### ISSUE THREAD

**Issue by paswan-ms** [OP-TEE/optee_os#372]
Hi,
In a TA we're writing we have a requirement for PKCS7 certificate handling. We've noticed that certificate handling doesn't appear to be part of the GP API surface and are thus attempting to build in such support from OpenSSL (using the OpenSSL configuration from Linaro/EDK2) into the TA. As far as I can tell so far, this port worked OK in EDK2 because the linker happily ignored everything that wasn't used (called). In the TA build, however, the linker is attempting to resolve everything that's creating a large number of unresolved references in functions that are never called. Does anyone know if it's possible to adjust the TA LD flags or .lds in a way that links via usage references?
Thanks,
Paul.

**Comment by ljmblueday** [comment]
Hi Paul.
Did you resolve this problem? I want to use openssl library into the TA, but I also get same problem. If you already resolve this problem, please advice for me.

Best regards,
Edward.

**Comment by paswan-ms** [comment]
I resolved it in so far as that I managed to get it all fully linked at least. It's still somewhat on the bloated side but I've not looked further into that yet to optimize it a bit more by using the OpenSSL flags to exclude the portions not required in this application. To get it into a TA I used the EDKII patch and ported over the remaining functionality from the EDKII CryptoPkg and EDKII system support functions that were missing from the OpTEE TA environment. 
Paul.
